### PR TITLE
[feat] Changing the layout with a single workspace view

### PR DESCRIPTION
**Pull request by emettely**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
Related to updates done in Storybook - https://bbc.github.io/digital-paper-edit-storybook/?path=/story/workspace-demo-only-not-published-on-npm--split-view

https://app.zeplin.io/project/5ce554083480691e05af2f74/screen/5dad8c523cd0612c5dfa3b55

**Describe what the PR does**    
See https://github.com/bbc/digital-paper-edit-firebase/pull/173#pullrequestreview-629255560 :)

**State whether the PR is ready for review or whether it needs extra work**    

**Additional context**

### CODE REVIEW

File: src/Components/Workspace/PaperEdits/index.js [hunk 1]
```diff
@@ -1,95 +1,54 @@
-import React, { useEffect, useState } from 'react';
-import ItemsContainer from '../../lib/ItemsContainer';
+import React from 'react';
 import PropTypes from 'prop-types';
-import Collection from '../../Firebase/Collection';
+// import Collection from '../../Firebase/Collection';
```
Comment (emettely) [inline src/Components/Workspace/PaperEdits/index.js:3]:
```suggestion
```

File: src/Components/Workspace/Transcripts/index.js [hunk 2]
```diff
@@ -1,214 +1,64 @@
...
 Transcripts.propTypes = {
-  projectId: PropTypes.any,
   firebase: PropTypes.any,
-  trackEvent: PropTypes.func
+  handleDeleteItem: PropTypes.any,
+  handleEditItem: PropTypes.any,
+  items: PropTypes.shape({
+    length: PropTypes.number,
+    map: PropTypes.func
+  }),
```
Comment (emettely) [inline src/Components/Workspace/Transcripts/index.js:58]:
```suggestion
  items: PropTypes.any,
```

**Review by emettely — COMMENTED** [review]
It looks like a lot but a lot of it was fixing warnings from React around useEffect. The core bulk of the things you'd want to look at are in `src/Components/Workspace/*` and `src/Components/Projects/index.js`, and some utility functions in `src/Util/*`.


Feature:
- [x] updates dependency for DPE-storybook, to use project row and transcript row components
- [x] workspace v2 implemented - relevant code `src/Components/Workspace/*` and `src/Components/Projects/index.js`, and some utility functions in `src/Util/*`
- [x] updated tracking event labels and functions
- [x] "duplicate" functionality implemented for projects and paper edits
- [ ] search feature

Optimisations
- [x] fixed warnings in useEffect - Collection instances will trigger re-render
- [x] added proptypes to WithAnalytics and ElementsDropdown
- [x] created and updated fields are updated on document creation 
- [x] some refactoring done for naming functions - createDefaultLabel
- [x] simplification of transcript editor paths: previous transcript links have looked like: `.../transcript/$transcriptId/correct` - transcripts are created after this point will look like `.../transcript/$transcriptId` instead.

Some gotchas
 - in order to prevent errors, there's now two route paths that both point to the TranscriptEditor component. (see src/contants/routes.js)

File: src/Components/Firebase/Collection.js [hunk 3]
```diff
@@ -31,7 +31,8 @@ class Collection {
     try {
       const docRef = await this.collectionRef.add({
         ...data,
-        created: this.getServerTimestamp()
+        created: this.getServerTimestamp(),
```
Comment (allishultes) [inline src/Components/Firebase/Collection.js:34]: When I first create a new project, this appears as 'NA' - and if I duplicate a newly created project, both fields flicker with real values, and then revert back to NA. I'm not sure why this'd be a bug (everything works if I navigate away from the page, and then navigate back). 

Screenshot 2021-04-09 at 08 24 24

File: src/Routes.js [hunk 4]
```diff
@@ -36,6 +36,11 @@ const Routes = ({ authUser }) => {
         <Route exact path={ ROUTES.PROJECTS } component={ Projects } />
         <Route exact path={ ROUTES.WORKSPACE } component={ Workspace } />
         <Route exact path={ ROUTES.PAPER_EDITOR } component={ PaperEditor } />
+        <Route
```
Comment (allishultes) [inline src/Routes.js:39]: I wonder if it's worth adding and leaving a comment about why we need two paths? Saw your note about 'gotchas' but might confuse folks working on the project.

**Review by allishultes — APPROVED** [review]
This looks good to me, Eimi - just a small question about timestamps on project creation. Also just curious about the "converted from" field on the transcripts now - does that ever populate? Lol.

File: src/Components/Firebase/Collection.js (hunk 3 above)
Comment (emettely) [reply src/Components/Firebase/Collection.js:34]: Thanks! Looks like I wasn't actually returning all the fields when getting from firestore :)

File: src/Routes.js (hunk 4 above)
Comment (emettely) [reply src/Routes.js:39]: > This looks good to me, Eimi - just a small question about timestamps on project creation. Also just curious about the "converted from" field on the transcripts now - does that ever populate? Lol.

it does now 😄

**Review by sarahrainbow — APPROVED** [review]
love the new design – looks great 🥇

**Pull request merged**